CRM/ACL/BAO/ACLEntityRole.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+
+/**
+ *  Access Control AclRole.
+ */
+class CRM_ACL_BAO_ACLEntityRole extends CRM_ACL_DAO_ACLEntityRole {
+  public static $_entityTable = NULL;
+
+  /**
+   * Whitelist of possible values for the entity_table field
+   *
+   * @return array
+   */
+  public static function entityTables(): array {
+    if (!self::$_entityTable) {
+      self::$_entityTable = [
+        'civicrm_contact' => ts('Contact'),
+        'civicrm_group' => ts('Group'),
+      ];
+    }
+    return self::$_entityTable;
+  }
+
+  /**
+   * @param array $params
+   *
+   * @return CRM_ACL_DAO_EntityRole
+   */
+  public static function create(&$params) {
+    $dao = new CRM_ACL_DAO_EntityRole();
+    $dao->copyValues($params);
+    $dao->save();
+    return $dao;
+  }
+
+  /**
+   * @param array $params
+   * @param $defaults
+   */
+  public static function retrieve(&$params, &$defaults) {
+    CRM_Core_DAO::commonRetrieve('CRM_ACL_DAO_EntityRole', $params, $defaults);
+  }
+
+  /**
+   * Update the is_active flag in the db.
+   *
+   * @param int $id
+   *   Id of the database record.
+   * @param bool $is_active
+   *   Value we want to set the is_active field.
+   *
+   * @return bool
+   *   true if we found and updated the object, else false
+   */
+  public static function setIsActive($id, $is_active) {
+    return CRM_Core_DAO::setFieldValue('CRM_ACL_DAO_EntityRole', $id, 'is_active', $is_active);
```

Comment:
It would be nice to clean up dangling references to the old DAO name, especially since this is a brand new file it ought to use the new names.